src/arch/riscv/tlb.cc

```diff
@@ -262,6 +264,21 @@ TLB::createPagefault(Addr vaddr, Mode mode)
     return std::make_shared<AddressFault>(vaddr, code);
 }
 
+// TODO: merge createPagefault and createAddrfault
+Fault
+TLB::createAddrfault(Addr vaddr, Mode mode)
+{
+    ExceptionCode code;
+    if (mode == TLB::Read) {
+        code = ExceptionCode::LOAD_ACCESS;
+    } else if (mode == TLB::Write) {
+        code = ExceptionCode::STORE_ACCESS;
+    } else {
```

Comment:
Is it true that Mode is one of these three? I suggest
```suggestion
    } else if (mode == TLB::Execute) {
```

Reply:
Yes, Mode is an enum with three possible values read, write, execute